docs/design/configs.md

```diff
@@ -0,0 +1,130 @@
+* Hard to change dynamically as needed by continuous services.
+* Hard to change in different versions that may support different configuration options.
+
+## Solution
+
+A new ``Config`` resource is proposed to solve the aforementioned problems 
+that is stored in ``etcd`` and distributed to interested/necessary components 
+through the API server. ``Config`` resource will be mounted as a new volume 
+and will be managed by a side-car container that communicates with the API server.
+
+The ``Config`` resource architecture will be very similar to that of ``Secrets``.
+However, the ``Config`` resource will be dynamic, meaning that different components consuming
+the same ``Config`` will be notified about the changes to that particular resource.
+
+### Advantages
+
+* Reusable across different components.
+* Easy distribution through the API server.
+* By leveraging the power of ``/watch API`` along with the new resource, we can change component configurations dynamically.
+* Layer of abstraction that gets rid of the global state currently used for command-line flags and environment variables.
+
+### API Resource
+
+A new resource for ``Config`` will be added to the ``API``:
+
+```go
+	type Config struct {
+		TypeMeta   `json:",inline"`
+		ObjectMeta `json:"metadata,omitempty"`
+
+		// Flags contains command-line flags.
+		Flags pflag.FlagSet `json:"flags,omitempty"`
+
+		// Variables contains environment variables.
+		Env []EnvVar `json:"env,omitempty"`
+	}
+
+	type ConfigList struct {
+		TypeMeta `json:",inline"`
+		ListMeta `json:"metadata,omitempty"`
+
+		Items []Config `json:"items"`
+	}
+```
+``Registry`` interface will be added to accompany the ``Config`` resource.  
+``Config`` information will be stored in ``etcd`` by default.
+
+### Validation
+
+``Flags`` and ``Env`` are planned to be validated by type-checking at the moment for types provided
+beforehand for planned configuration parameters (``maxPodNumber`` => ``int``).
+
+### Volume Source
+
+A new ``ConfigSource`` type of volume source containing the ``Config`` object will be added to the ``VolumeSource`` struct in the API:
+
+```go
+	type ConfigVolumeSource struct {
+        ConfigName string `json:"configName"`
+	}
+```
+
+### Configuration
+
+Configured component (with ``Kubectl`` or possibly other means) will send it's configuration to the API server.
+
+```json
+{
+    "apiVersion": "v1beta3",
+    "kind": "Config",
+    "id": "",
+    "flags": {
+        "api_server": "132.32.13.14:8998",
+        "api_version": "v1beta3",
+    },
+    "variables": {
+        "DB_HOST": "1.1.1.1",
+        "DB_PORT": "8778",
+    }
+}
+```
+
+The owner of the configuration will be able mutate the configuration and it will be immutable from the
+perspective of other components (**Note:** A ``ConfigController`` might be added to undertake this responsibility).
+
+Consumers will have a container listening to the API server for configuration changes and reload them after a change.
```

Comment:
I think a volume source is more consistent with our other choices, and if you want to reload based on config that seems either an option you handle by watching for file changes (more decoupled) or by asking.  Can always be a second tier implementation point